ckanext/canada/commands.py

```diff
@@ -313,7 +316,30 @@ def _changed_packages_since(self, registry, since_time):
         for result in data:
             package_id = result['data']['package']['id']
             try:
-                packages.append(json.dumps(registry.action.package_show(id=package_id)))
+                source_package = registry.action.package_show(id=package_id)
+                for source_resource in source_package['resources']:
+                    # check if resource exists in datastore
+                    source_table = registry.call_action('datastore_search',
+                                                        {'id': source_resource['id'], 'limit': 0})
+                    if source_table:
+                        from ckanext.datastore.helpers import datastore_dictionary
+                        import simplejson
+
+                        # fetch all records
+                        source_records = registry.call_action('datastore_search',
+                                                              {'id': source_resource['id'],
+                                                               'limit': source_table['total']})
```

Comment:
pulling the records like this won't work for larger tables (need to implement pagination), it's also pretty slow. Was shelling out to pg_dump too complicated?

Reply:
if we don't want to shell out to pg_dump for copying the data we could run a manual xloader job after running portal_update instead (launch the jobs worker process but tell it to exit on completion)